packages/telemetry/src/span-exporters/cloud-span-exporter.ts

```diff
@@ -50,6 +54,11 @@ export class OTLPTraceExporter extends OTLPTraceExporterHttp {
    */
   attachProjectId (projectId: string | null | undefined): void {
     if (!projectId) {
+      if (this.requirementsToExport === 'unknown') {
+        this.requirementsToExport = 'unmet'
```

Comment:
why do we need unknown and unmet? can we assume 'unmet' until met?

Reply:
any reason this wasn't a bool?

Reply:
when encryption is on, we can't assume unmet until we get a null project id or record key. The send function will behave differently. If unmet we error instantly, if unknown we will delay the request.

Reply:
Otherwise we could incorrectly fail any spans sent prior to getting the project id/record key, since we capture spans prior to them being available.

Reply:
sure